ReSharper.FSharp/src/FSharp.Psi.Features/src/Internal/FSharpInternalActionGroup.fs

```diff
@@ -42,12 +45,37 @@ type DumpCurrentFcsProjectAction() =
                     match fcsProjectProvider.GetFcsProject(psiModule) with
                     | Some fcsProject -> writer.WriteLine(fcsProject.TestDump(writer))
                     | _ -> ())
+            
+            
+[<Action("FSharp_Internal_DumpCurrentFcsFile", "Dump current FCS file")>]
+type DumpCurrentFcsFileAction() =
+
+    [<DefaultValue>]
+    val mutable myTextControl: JetBrains.TextControl.ITextControl
+
+    interface IExecutableAction with
+        member this.Update(context, _, _) =
+            isNotNull (context.GetData(ProjectModelDataConstants.PROJECT))
+
+        member this.Execute(context, _) =
+            let editorContext = context.GetData(JetBrains.DocumentModel.DataContext.DocumentModelDataConstants.EDITOR_CONTEXT)
+            if isNull editorContext then () else
+
+            let sourceFile = context.GetData(JetBrains.ReSharper.Psi.DataContext.PsiDataConstants.SOURCE_FILE)
+            let file = if isNull sourceFile then null else sourceFile.GetPsiFile(editorContext.CaretOffset)
```

Comment:
Please use `PsiFilesExtensions.GetPrimaryPsiFile`, and there will be no need to pass specific offset/context.

The offset is relevant when there're injected languages are used, and you may be interested in a file in another language that is nested inside some other node.

Reply:
Do we want to enable it for F# files only?

If yes, it would make sense to check the file language here.
If not, then I should probably move this action to the Rider codebase instead, but till then it'd still be nice to have it available here.

Reply:
Mmh, from my point of view there's no reason to restrict it to F# only.
So, if you could move this action at some point to the Rider codebase, that would be nice.